### DISCUSSION TITLE
Help with migration connection error

### DISCUSSION THREAD

**Discussion by alongthewayhome** [adonisjs/core#3755] [Help]
When attempting to use Adonis migration:run, I end up having difficulties with a connection error. 

`Error: connect ECONNREFUSED 127.0.0.1:3306
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)
    --------------------
    at Protocol._enqueue (/mnt/c/projects/sneakers_ims/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Protocol.handshake (/mnt/c/projects/sneakers_ims/node_modules/mysql/lib/protocol/Protocol.js:51:23)
    at Connection.connect (/mnt/c/projects/sneakers_ims/node_modules/mysql/lib/Connection.js:116:18)
    at /mnt/c/projects/sneakers_ims/node_modules/knex/lib/dialects/mysql/index.js:68:18
    at new Promise (<anonymous>)
    at Client_MySQL.acquireRawConnection (/mnt/c/projects/sneakers_ims/node_modules/knex/lib/dialects/mysql/index.js:63:12)
    at create (/mnt/c/projects/sneakers_ims/node_modules/knex/lib/client.js:290:39) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 3306,
  fatal: true
}`

I have made sure the port is on and hosting my database as well as other little things like switching port numbers with no luck. Would anyone have an idea on where I am going wrong? Here is my .env file as well.

`HOST=127.0.0.1
PORT=3333
NODE_ENV=development
APP_URL=http://${HOST}:${PORT}
CACHE_VIEWS=false
APP_KEY=SAGWrGnmKlX4gLdeuDqArGSB1KE3YEiF
DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_USER=root
DB_PASSWORD=root
DB_DATABASE=adonis
SESSION_DRIVER=cookie
HASH_DRIVER=bcrypt`

**Comment by WeyIin** [comment]
`Error: connect ECONNREFUSED 127.0.0.1:3306` 

Something is telling me that your DB is not listening to this address / port. 

Make sure your DB is UP and running
Double check address AND port
Instead of `127.0.0.1` try `localhost`

**Reply by alongthewayhome** [reply]
I think so too. I made sure DB is up and running while checking address and port number with no luck. I also tried replacing 127.0.0.1 with localhost with no luck. I'm posting a picture of my database server status in case there are any clues that can help with resolving the connection error.
![Screenshot (18)](https://user-images.githubusercontent.com/106983844/173671213-0c540df0-6a33-466a-a28e-f07f41695d5a.png)

**Reply by WeyIin** [reply]
Check database configuration as well in `config` and make sure it looks something like this

```
    mysql: {
      client: 'mysql',
      connection: {
        host: Env.get('MYSQL_HOST'),
        port: Env.get('MYSQL_PORT'),
        user: Env.get('MYSQL_USER'),
        password: Env.get('MYSQL_PASSWORD', ''),
        database: Env.get('MYSQL_DB_NAME'),
      },
      healthCheck: false,
      debug: false,
      seeders: {
        paths: ['./database/seeders/main']
      }
    },
```

**Reply by alongthewayhome** [reply]
The database configuration is identical, but still no luck. I did notice an issue just now. When typing localhost into my browser, it directs me to the index of my server. But typing in localhost:3306 gives me a broken page. Could this indicate the reason behind my connection error?

**Reply by WeyIin** [reply]
try these `.env` settings

```
MYSQL_HOST=localhost
MYSQL_PORT=3306
MYSQL_USER=root
MYSQL_PASSWORD=root
MYSQL_DB_NAME=adonis
PORT=3333
HOST=0.0.0.0
NODE_ENV=development
```

also edit: access your page via `127.0.0.1:3333`

**Reply by alongthewayhome** [reply]
I tried the .env settings you recommend, but still no luck.

**Reply by WeyIin** [reply]
Try connecting to the db via command line, just to check if it's something with your app or connection to database.

Actually go through all these steps https://phoenixnap.com/kb/connect-to-mysql-windows-command-line

see if mysql is actually running, and then connect via cmd.

`mysql.exe -u root -h localhost -p`

**Reply by alongthewayhome** [reply]
I was able to connect to mysql through cmd and followed all the steps you recommend, but am still receiving a connection error when attempting migration:run.

**Reply by WeyIin** [reply]
Do you have correct driver installed? What is the mysql version? It could be that you're using mysql driver instead of mysql2? or vice versa?

**Reply by alongthewayhome** [reply]
Mysql version 8.0.29. I installed the MySQL Connector/Node.js driver on this site: https://dev.mysql.com/doc/dev/connector-nodejs/8.0/
But still I am receiving a connection error.

**Reply by WeyIin** [reply]
What mysql do you have installed under package.json?

**Reply by alongthewayhome** [reply]
```javascript
"mysql": "^2.18.1"
```

**Comment by thetutlage** [comment]
Can you share the `config/database.ts` file and also the `.env` file? The final one's you are using

**Reply by alongthewayhome** [reply]
Here is the final config/database.ts file I am using.
```javascript
'use strict'

/** @type {import('@adonisjs/framework/src/Env')} */
const Env = use('Env')

/** @type {import('@adonisjs/ignitor/src/Helpers')} */
const Helpers = use('Helpers')

module.exports = {
  /*
  |--------------------------------------------------------------------------
  | Default Connection
  |--------------------------------------------------------------------------
  |
  | Connection defines the default connection settings to be used while
  | interacting with SQL databases.
  |
  */
  connection: Env.get('DB_CONNECTION', 'mysql'),

  /*
  |--------------------------------------------------------------------------
  | Sqlite
  |--------------------------------------------------------------------------
  |
  | Sqlite is a flat file database and can be good choice under development
  | environment.
  |
  | npm i --save sqlite3
  |
  */
  sqlite: {
    client: 'sqlite3',
    connection: {
      filename: Helpers.databasePath(`${Env.get('DB_DATABASE', 'development')}.sqlite`)
    },
    useNullAsDefault: true
  },

  /*
  |--------------------------------------------------------------------------
  | MySQL
  |--------------------------------------------------------------------------
  |
  | Here we define connection settings for MySQL database.
  |
  | npm i --save mysql
  |
  */
  mysql: {
    client: 'mysql',
    connection: {
      host: Env.get('DB_HOST', 'localhost'),
      port: Env.get('DB_PORT', ''),
      user: Env.get('DB_USER', 'root'),
      password: Env.get('DB_PASSWORD', ''),
      database: Env.get('DB_DATABASE', 'adonis')
    }
  },
  

  /*
  |--------------------------------------------------------------------------
  | PostgreSQL
  |--------------------------------------------------------------------------
  |
  | Here we define connection settings for PostgreSQL database.
  |
  | npm i --save pg
  |
  */
  pg: {
    client: 'pg',
    connection: {
      host: Env.get('DB_HOST', 'localhost'),
      port: Env.get('DB_PORT', ''),
      user: Env.get('DB_USER', 'root'),
      password: Env.get('DB_PASSWORD', ''),
      database: Env.get('DB_DATABASE', 'adonis')
    }
  }
}

```

Here is the final .env file I am using.
```
HOST=127.0.0.1
PORT=3333
NODE_ENV=development
APP_URL=http://${HOST}:${PORT}
CACHE_VIEWS=false
APP_KEY=SAGWrGnmKlX4gLdeuDqArGSB1KE3YEiF
DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_USER=root
DB_PASSWORD=root
DB_DATABASE=adonis
SESSION_DRIVER=cookie
HASH_DRIVER=bcrypt
```

**Reply by bunkerapps** [reply]
have you already created a DB called "adonis" on your MYSQL?

**Reply by ewertonrfd** [reply]
> Here is the final config/database.ts file I am using.
> 
> ```js
> 'use strict'
> 
> /** @type {import('@adonisjs/framework/src/Env')} */
> const Env = use('Env')
> 
> /** @type {import('@adonisjs/ignitor/src/Helpers')} */
> const Helpers = use('Helpers')
> 
> module.exports = {
>   /*
>   |--------------------------------------------------------------------------
>   | Default Connection
>   |--------------------------------------------------------------------------
>   |
>   | Connection defines the default connection settings to be used while
>   | interacting with SQL databases.
>   |
>   */
>   connection: Env.get('DB_CONNECTION', 'mysql'),
> 
>   /*
>   |--------------------------------------------------------------------------
>   | Sqlite
>   |--------------------------------------------------------------------------
>   |
>   | Sqlite is a flat file database and can be good choice under development
>   | environment.
>   |
>   | npm i --save sqlite3
>   |
>   */
>   sqlite: {
>     client: 'sqlite3',
>     connection: {
>       filename: Helpers.databasePath(`${Env.get('DB_DATABASE', 'development')}.sqlite`)
>     },
>     useNullAsDefault: true
>   },
> 
>   /*
>   |--------------------------------------------------------------------------
>   | MySQL
>   |--------------------------------------------------------------------------
>   |
>   | Here we define connection settings for MySQL database.
>   |
>   | npm i --save mysql
>   |
>   */
>   mysql: {
>     client: 'mysql',
>     connection: {
>       host: Env.get('DB_HOST', 'localhost'),
>       port: Env.get('DB_PORT', ''),
>       user: Env.get('DB_USER', 'root'),
>       password: Env.get('DB_PASSWORD', ''),
>       database: Env.get('DB_DATABASE', 'adonis')
>     }
>   },
>   
> 
>   /*
>   |--------------------------------------------------------------------------
>   | PostgreSQL
>   |--------------------------------------------------------------------------
>   |
>   | Here we define connection settings for PostgreSQL database.
>   |
>   | npm i --save pg
>   |
>   */
>   pg: {
>     client: 'pg',
>     connection: {
>       host: Env.get('DB_HOST', 'localhost'),
>       port: Env.get('DB_PORT', ''),
>       user: Env.get('DB_USER', 'root'),
>       password: Env.get('DB_PASSWORD', ''),
>       database: Env.get('DB_DATABASE', 'adonis')
>     }
>   }
> }
> ```
> 
> Here is the final .env file I am using.
> 
> ```
> HOST=127.0.0.1
> PORT=3333
> NODE_ENV=development
> APP_URL=http://${HOST}:${PORT}
> CACHE_VIEWS=false
> APP_KEY=SAGWrGnmKlX4gLdeuDqArGSB1KE3YEiF
> DB_CONNECTION=mysql
> DB_HOST=127.0.0.1
> DB_PORT=3306
> DB_USER=root
> DB_PASSWORD=root
> DB_DATABASE=adonis
> SESSION_DRIVER=cookie
> HASH_DRIVER=bcrypt
> ```

And you will edit only the file .env, no the file database.ts
